migrate/list-convert.go

```diff
@@ -0,0 +1,259 @@
+	"k8s.io/klog/v2"
+)
+
+const (
+	// DriveTypeMetaKind is for the kind
+	DriveTypeMetaKind = "DirectPVDrive"
+	// VolumeTypeMetaKind is for the volume kind
+	VolumeTypeMetaKind = "DirectPVVolume"
+	// APIVersion is for the version
+	APIVersion = "directpv.min.io/v1beta1"
+	// DirectCSIGroup is for the group
+	DirectCSIGroup = "direct.csi.min.io"
+)
+
+var (
+	directCSIDriveClient  typeddirectcsi.DirectCSIDriveInterface
+	directCSIVolumeClient typeddirectcsi.DirectCSIVolumeInterface
+	uuidRegex             = regexp.MustCompile("^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$")
+)
+
+func init() {
+	k8s.Init()
+
+	var err error
+	if directCSIDriveClient, err = directcsiclient.DirectCSIDriveInterfaceForConfig(k8s.KubeConfig()); err != nil {
+		klog.Fatalf("unable to create new direct-csi drive interface; %v", err)
+	}
+
+	if directCSIVolumeClient, err = directcsiclient.DirectCSIVolumeInterfaceForConfig(k8s.KubeConfig()); err != nil {
+		klog.Fatalf("unable to create new direct-csi volume interface; %v", err)
+	}
+
+	gvk, err := directcsiclient.GetGroupKindVersions(
+		DirectCSIGroup,
+		"DirectCSIDrive",
+		"v1beta5",
+		directv1beta4.Version,
+		directv1beta3.Version,
+		directv1beta2.Version,
+		directv1beta1.Version,
+		directv1alpha1.Version,
+	)
+	if err != nil {
+		klog.Fatalf("unable to get GroupKindVersions of direct-csi; %v", err)
+	}
+
+	if gvk.Group != DirectCSIGroup {
+		klog.Fatalf("migration does not support direct-csi group %v", gvk.Group)
+	}
+
+	switch gvk.Version {
+	case "v1beta3", "v1beta4", "v1beta5":
+	default:
+		klog.Fatalf("migration does not support direct-csi version %v", gvk.Version)
+	}
+}
+
+// Preparing a map[driveName]fsuuid
+var driveFsuuid = make(map[string]string)
+
+func migrateVolumeCRD(oldVolumeList []directcsi.DirectCSIVolume) (newVolumeList []directpv.DirectPVVolume, theerror error) {
+	newVolumes := []directpv.DirectPVVolume{}
+	for _, OldVolume := range oldVolumeList {
+		// FIXME: use types.NewDrive() instead of creating structure.
+		NewVolume := directpv.DirectPVVolume{}
+
+		// TypeMeta fields
+		NewVolume.TypeMeta.Kind = VolumeTypeMetaKind
+		NewVolume.TypeMeta.APIVersion = APIVersion
+
+		// ObjectMeta fields
+		NewVolume.ObjectMeta = OldVolume.ObjectMeta
+
+		// Status fields
+		NewVolume.Status.DataPath = OldVolume.Status.HostPath
+		NewVolume.Status.StagingTargetPath = OldVolume.Status.StagingPath
+		NewVolume.Status.TargetPath = OldVolume.Status.ContainerPath
+		NewVolume.Status.FSUUID = driveFsuuid[OldVolume.Status.Drive]
+		NewVolume.Status.TotalCapacity = OldVolume.Status.TotalCapacity
+		NewVolume.Status.AvailableCapacity = OldVolume.Status.AvailableCapacity
+		NewVolume.Status.UsedCapacity = OldVolume.Status.UsedCapacity
+
+		// Append the new drive to the list
+		newVolumes = append(newVolumes, NewVolume)
+	}
+	return newVolumes, nil
+}
+
+func migrateDriveCRD(oldDriveList []directcsi.DirectCSIDrive) (newDriveList []directpv.DirectPVDrive, therror error) {
```

Comment:
i think this is a duplicate. We have this function already in `cmd/kubectl-directpv/migrate.go`